src/gasman.c

```diff
@@ -2081,54 +2081,68 @@ void CheckMasterPointers( void )
 }
 
 
-/****************************************************************************
-**
-*F  SwapMasterPoint( <bag1>, <bag2> ) . . . swap pointer of <bag1> and <bag2>
-*/
-void SwapMasterPoint (
-    Bag                 bag1,
-    Bag                 bag2 )
+// Swap the master pointers of bag1 and bag2
+//
+// We need to make sure the correct bags are garbage collected, so we always put
+// *both* bags on the ChangedBags linked-list, rather than pick through the
+// exact cases, as it is never incorrect to mark something changed.
+//
+// For completeness and future reference here are the necessary points to
+// consider.
+//
+// As ChangedBags is a singly linked list, we can't easily take anything out of
```

Comment:
At this point, reader of this comment have no idea why they should care about removing something from the ChangedBags list. So I think this whole explanation is a bit backwards.